### PR TITLE
PS-1582 Modify timezone to UTC while accepting and in api response

### PR DESCRIPTION
**Pull request by Xitija**
## Summary by CodeRabbit

- **New Features**
	- Enhanced date validation for event creation, ensuring accurate checks for recurring events.
	- Introduced new properties in data models, including `providerGenerated` and `autoEnroll`, improving event management capabilities.
	- Consolidated event creation logic for online and offline events into a single method.
	- Added new methods for deleting events and managing partially created data.

- **Bug Fixes**
	- Improved error messaging for date validation to enhance user clarity.

- **Documentation**
	- Updated validation logic descriptions to reflect new rules and properties for event creation.

### CODE REVIEW

File: src/common/pipes/event-validation.pipe.ts [hunk 1]
```diff
@@ -1,63 +1,50 @@
...
-    const endDate = getTimezoneDate(
-      timeZone,
-      new Date(createEventDto.endDatetime),
-    );
-    const currentDate = getTimezoneDate(timeZone); // Current date in the specified timezone
+    const eventStartDate = createEventDto.startDatetime.split('T')[0];
+    const eventEndDate = createEventDto.endDatetime.split('T')[0];
 
-    if (startDate <= currentDate) {
+    const startDate = new Date(createEventDto.startDatetime);
```
Comment (vaivk369) [inline src/common/pipes/event-validation.pipe.ts:16]: new Date(createEventDto.startDatetime); will give you date from the parameter right? 
Why     const eventStartDate = createEventDto.startDatetime.split('T')[0]; this code is written?

Comment (vaivk369) [reply src/common/pipes/event-validation.pipe.ts:16]: @Xitija

File: src/common/utils/constants.util.ts [hunk 2]
```diff
@@ -16,10 +18,19 @@ export const ERROR_MESSAGES = {
   REGISTRATION_END_DATE_INVALID: 'Registration end date must be in the future',
   REGISTRATION_START_DATE_BEFORE_END_DATE:
     'Registration start date must be before registration end date',
-  RECURRENCE_END_DATE_INVALID: 'Recurrence end date must be in the future',
-  RECURRENCE_END_DATE_BEFORE_EVENT_DATE:
+  RECURRENCE_END_DATE_INVALID:
+    'Recurrence end date must be a valid ISO 8601 date string ',
```
Comment (snehal0904) [inline src/common/utils/constants.util.ts:22]: @Xitija @vaivk369 why we need ISO 8601 in error message, this will create confusion for end user

Comment (Xitija) [reply src/common/utils/constants.util.ts:22]: Standard in javascript [](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#date_time_string_format)

**Pull request merged**